Release broken due to Playwright report artifacts

From https://github.com/kiesraad/abacus/actions/runs/13354824772/job/37296670441:

```
Run sha256sum -b * > SHA256SUMS
  sha256sum -b * > SHA256SUMS
  shell: /usr/bin/bash -e {0}
  env:
    BUILD_DATE: 2025-02-16
    TAG: v0.1.0-alpha.20250216.06cbfae
sha256sum: playwright-report-release-macos-latest: Is a directory
sha256sum: playwright-report-release-ubuntu-latest: Is a directory
sha256sum: playwright-report-release-windows-latest: Is a directory
Error: Process completed with exit code 1.
```

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -125,6 +125,7 @@ jobs:
       - name: Download abacus
         uses: actions/download-artifact@v4
         with:
+          pattern: abacus-*
           path: release/
       - name: Rename binary
         run: |

